src/applications/find-va-forms/containers/SearchResults.jsx

```diff
@@ -1,92 +1,156 @@
-  }
-
-  if (!results) {
-    return null;
-  }
-
-  if (!results.length) {
-    return (
-      <h2 className="vads-u-font-size--lg vads-u-margin-top--1p5 vads-u-font-weight--normal">
-        No results found for "<strong>{query}</strong>"
-      </h2>
-    );
-  }
-
-  return (
-    <>
-      <h2 className="vads-u-font-size--lg vads-u-margin-top--1p5 vads-u-font-weight--normal">
-        Showing results for "<strong>{query}</strong>"
-      </h2>
-
-      <SortableTable
-        className="vads-u-margin--0"
-        currentSort={{ order: 'ASC', value: 'title' }}
-        data={results}
-        fields={tableFields}
-      />
-    </>
-  );
-};
-
-SearchResults.propTypes = {
-  // From mapStateToProps.
-  fetching: PropTypes.bool.isRequired,
-  query: PropTypes.string.isRequired,
-  results: PropTypes.arrayOf(
-    PropTypes.shape({
-      id: PropTypes.string.isRequired,
-      type: PropTypes.string.isRequired,
-      attributes: PropTypes.shape({
+class SearchResults extends Component {
+  static propTypes = {
+    // From mapStateToProps.
+    fetching: PropTypes.bool.isRequired,
+    query: PropTypes.string.isRequired,
+    results: PropTypes.arrayOf(
+      PropTypes.shape({
+        // Original form data key-value pairs.
         firstIssuedOn: PropTypes.string.isRequired,
         formName: PropTypes.string.isRequired,
+        id: PropTypes.string.isRequired,
         lastRevisionOn: PropTypes.string.isRequired,
         pages: PropTypes.number.isRequired,
         sha256: PropTypes.string.isRequired,
         title: PropTypes.string.isRequired,
+        type: PropTypes.string.isRequired,
         url: PropTypes.string.isRequired,
+        // Table field labels that can be JSX.
+        idLabel: PropTypes.node.isRequired,
+        titleLabel: PropTypes.node.isRequired,
+        descriptionLabel: PropTypes.node.isRequired,
+        availableOnlineLabel: PropTypes.node.isRequired,
       }).isRequired,
-    }).isRequired,
-  ),
-};
+    ),
+    // From mapDispatchToProps.
+    updateResults: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedFieldLabel: 'idLabel',
+      selectedFieldOrder: 'ASC',
+    };
+  }
+
+  onHeaderClick = (fieldLabel, order) => {
+    const { selectedFieldLabel, selectedFieldOrder } = this.state;
+
+    // Derive the opposite sort order.
+    const oppositeFieldOrder = selectedFieldOrder === 'ASC' ? 'DESC' : 'ASC';
+
+    // Sort the results.
+    this.sortResults(
+      fieldLabel,
+      fieldLabel === selectedFieldLabel ? oppositeFieldOrder : order,
+    );
+  };
+
+  sortResults = (fieldLabel, selectedFieldOrder = 'ASC') => {
+    const { results, updateResults } = this.props;
+
+    // Update local state for SortableTable.
+    this.setState({ selectedFieldLabel: fieldLabel, selectedFieldOrder });
+
+    // Derive the original field (not the JSX label).
+    const field = fieldLabel.replace('Label', '');
```

Comment:
Confusing...the `Label` part is because we're passing JSX into the `SortableTable`?

Reply:
Ya I don't like how this is set up. Basically let's say we have a field called `'blah'`, then the corresponding JSX label for `<SortableTable />` would be `'blahLabel'` as the original `'blah'` field is also on that same object. Basically for each field we have 1) its value and 2) a corresponding key-value pair that is JSX for `<SortableTable />`